shared/utility/io/uart.hpp

```diff
@@ -43,7 +43,14 @@ namespace utility::io {
          * @brief We can't copy these because otherwise we might close the device twice
          */
         uart(const uart& uart) = delete;
-        uart(uart&&)           = default;
+        uart& operator=(const uart& uart) = delete;
+
+        /**
+         * @brief We can move these because it won't close the device twice
+         */
+        uart(uart&&)  = default;
+        uart& operator=(uart&& uart) = default;
```

Comment:
I'm not sure we can move them... would the other instance no longer have something to close?
We might need explicit move constructors and operators